src/plugins/management/public/components/management_app/management_app.tsx

```diff
@@ -79,8 +82,19 @@ export const ManagementApp = ({ dependencies, history }: ManagementAppProps) =>
   );
 
   useEffect(() => {
+    // If workspace is enabled and user has entered workspace, hide advance settings and dataSource menu
+    if (capabilities.workspaces.enabled && workspaces.currentWorkspace$.getValue()) {
```

Comment:
Is it safely to access `capabilities.workspaces.enabled`? what if workspaces is undefined/null?

Reply:
As in type definition workspaces capabilities is required. Should we update?

Reply:
Just a call out, I am just wondering if workspaces will still be a object if workspaces feature flag is turned off.

Reply:
Yes. I checked that there are some other places using this way. https://github.com/ruanyl/OpenSearch-Dashboards/pull/322/files#diff-47d724c5a7bb6e8c1286f67e3ed1b117ac27e05387052760388bd477bd76337bL184
I am thinking that should we maintain to keep `capabilities.workspaces.enabled` existing and its value should be false if workspace is turned off?

Reply:
BTW, if we put this logic in workspace plugin, we don't need to judge if workspace is enabled.